src/components/Tooltip.tsx

```diff
@@ -51,10 +51,33 @@ function Tooltip({
         return child
     }
 
+    /**
+     * Prevents the tooltip from automatically firing on focus all the time. This is to prevent
+     * tooltips from showing when the trigger element is focused back after a popover or dialog that
+     * it opened was closed. See link below for more details.
+     * @see https://github.com/reakit/reakit/discussions/749
+     */
+    function handleFocus(event: React.FocusEvent) {
+        // If focus is not followed by a key up event, does it mean that it's not
+        // an intentional keyboard focus? Not sure but it seems to work.
+        function handleKeyUp(e: Event) {
+            if ((e as KeyboardEvent).key !== 'Escape') tooltip.show()
```

Comment:
This will cover the use case with the `Escape` key, but if a menu item is activated with an `Enter` or `Space` key the same problem will persist. 

I was thinking we could probably show/hide the tooltip if the focus event was caused by a user or programatically, but it looks like React's `FocusEvent` does contain `originalEvent`. Is there a way to check events like this in React?

Reply:
> This will cover the use case with the Escape key, but if a menu item is activated with an Enter or Space key the same problem will persist.

Yup. I realized this after the fact and left a comment in the linked conversation in the reakit repo. I figured it is acceptable as is because after all it affects keyboard-centric users but unblocks us from the main worrying group: mouse-centric users complaining about this.

However, if you have an idea and can take a shot at it, go for it. Just don't let this take too much of your time if you don't get lucky quickly. I can resume this after my time off if needed.